Angular/Angular guidelines and best practices/DatX data store.md

```diff
@@ -0,0 +1,467 @@
+import { InjectionToken } from '@angular/core';
+import { AppCollection } from '<path-to-collection-definition>';
+
+export const APP_COLLECTION = new InjectionToken<AppCollection>(
+  'APP_COLLECTION'
+);
+```
+
+```ts
+// app.module.ts
+import { APP_COLLECTION } from '<path-to-token-definition>';
+import { NgModule } from '@angular/core';
+import { AppComponent } from './app.component';
+import { AppCollection } from '<path-to-collection-definition>';
+
+@NgModule({
+	declarations: [AppComponent],
+	imports: [
+		...
+	],
+	providers: [
+    ...
+    {
+      provide: APP_COLLECTION,
+      useValue: new AppCollection();
+    }
+    ...
+  ],
+  bootstrap: [AppComponent],
+}
+export class AppModule {}
+```
+
+Then whenever you want to work with the collection from within Angular's DI container, you can simply use the token above to inject the collection instance.
+
+```ts
+// example.service.ts
+import { APP_COLLECTION } from '<path-to-token-definition>';
+import { Inject, Injectable } from '@angular/core';
+import { AppCollection } from '<path-to-collection-definition>';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExampleService {
+  constructor(
+    @Inject(APP_COLLECTION) protected readonly collection: AppCollection // app.module.ts scoped instance of AppCollection
+  ) {}
+}
+```
+
+When testing consumers of APP_COLLECTION, just inject the collection under the same token again as with any other dependency:
+
+```ts
+// example.service.spec.ts
+import { AppCollection } from '<path-to-collection-definition>';
+import { APP_COLLECTION } from '<path-to-token-definition>';
+import { TestBed } from '@angular/core/testing';
+import { ExampleService } from './example.service';
+
+describe('ExampleService', () => {
+  let service: ExampleService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: APP_COLLECTION,
+          useValue: new AppCollection(), // empty mock collection
+        },
+      ],
+    });
+    service = TestBed.inject(ExampleService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+});
+```
+
+## Configuring DatX HTTP calls interception
+
+By default, DatX uses Fetch API when invoking HTTP calls, but this means that it also bypass **HttpClient** from `@angular/common/http` and therefore any registered interception logic or similar kind of middleware. Luckily there is a way around that, see [official documentation guide](https://datx.dev/docs/jsonapi-angular/base-fetch).
+
+## Abstracting entity related login into separate service
+
+You might want to abstract away a little of the tediousness of working with DatX and/or bound some special logic to a given entity. Since many of the Collection's method require an Entity type as a parameter you can create separate service that will hide this small implementation detail for you. One way to do that would be to extend an abstract class similar to this one:
```

Comment:
```suggestion
You might want to abstract away a bit of the tediousness of working with DatX and/or bind some special logic to a given entity. Since many of the Collection's methods require an Entity type as a parameter you can create separate service that will hide this small implementation detail for you. One way to do that would be to extend an abstract class similar to this one:
```